BazelExtensions/xcode_configuration_provider.bzl

```diff
@@ -168,37 +172,43 @@ def _source_output_file_map(target, ctx):
     source_output_file_map = ctx.actions.declare_file("{}_source_output_file_map.json".format(target.label.name))
 
     mapping = {}
-    objc_srcs = []
-    objc_objects = []
+    srcs = []
+    objs = []
 
     # List of source files to be mapped to output files
     if hasattr(ctx.rule.attr, "srcs"):
-        objc_srcs = [
+        srcs = [
             f
             for source_file in ctx.rule.attr.srcs
             for f in source_file.files.to_list()
-            # Handling objc only for now
-            if f.path.endswith((".m", ".mm", ".c", ".cc", ".cpp"))
-            # TODO: handle swift
+            if f.path.endswith((".m", ".mm", ".c", ".cc", ".cpp", ".swift"))
         ]
 
     # Get compilation outputs if present
     if OutputGroupInfo in target:
+        # Objc
         if hasattr(target[OutputGroupInfo], "compilation_outputs"):
-            objc_objects.extend(target[OutputGroupInfo].compilation_outputs.to_list())
+            objs.extend([f.path for f in target[OutputGroupInfo].compilation_outputs.to_list()])
+        # Swift
+        if hasattr(target[OutputGroupInfo], "swift_ast_file"):
```

Comment:
Is this a dependency of this working E2E? AFAIK the AST dump feature is probably more obscure / less used so it might be tricky..  I imagined there was some other way to compute them: e.g. take the parent of the another output assume it has one for each source?

Reply:
Good point, let me confirm if this is guaranteed to be there or not. Need to dig into that ast PR. Will report back.

Reply:
Yeah, I don't think this is always there indeed. Will look for an alternative approach here. Thx for calling it out!

Reply:
SGTM - I haven't looked at it as much as you so it's mainly a quesiton:
- if it guaranteed output group but the output is never built - it feels a bit quirky but works fine since you _derive_ a value from it
- if if is not gaurenteed the probably strategy solution is in-order

Reply:
@jerrymarino I don't think this is guaranteed, found the PR that introduced this and there's checks to only do work if the ast flag is present. I think it's error prone to depend on this.

See new commit I just pushed here, collecting this directly from the action's outputs instead. Open to suggestions if there's a provider I should be using here.